hivemq-edge/src/main/java/com/hivemq/pulse/asset/PulseAgentAssetChangedListener.java

```diff
@@ -16,26 +16,34 @@
 
 package com.hivemq.pulse.asset;
 
+import com.hivemq.configuration.reader.AssetMappingExtractor;
 import com.hivemq.configuration.reader.PulseExtractor;
 import com.hivemq.pulse.utils.PulseAgentAssetUtils;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.List;
 
 public class PulseAgentAssetChangedListener implements ExternalAssetProvider.AssetChangedListener {
+    private final @NotNull AssetMappingExtractor assetMappingExtractor;
     private final @NotNull PulseExtractor pulseExtractor;
 
     public PulseAgentAssetChangedListener(
+            final @NotNull AssetMappingExtractor assetMappingExtractor,
             final @NotNull PulseExtractor pulseExtractor) {
+        this.assetMappingExtractor = assetMappingExtractor;
         this.pulseExtractor = pulseExtractor;
     }
 
+    public @NotNull AssetMappingExtractor getAssetMappingExtractor() {
+        return assetMappingExtractor;
+    }
+
     public @NotNull PulseExtractor getPulseExtractor() {
         return pulseExtractor;
     }
 
     @Override
     public void onAssetsChanged(final @NotNull List<Asset> remoteAssets) {
```

Comment:
I went down into the code invoked by this and there are blocking calls (replaceConfigsAndTriggerWrite) there.
Is it fine for the invoker to be blocked?

Reply:
Yes. I believe these changes have to be atomic, and there is a global lock ensuring this is atomic.